crates/rpc/rpc/src/eth/api/fees.rs

```diff
@@ -137,7 +137,7 @@ where
                 sorter.sort();
 
                 let mut rewards = Vec::with_capacity(reward_percentiles.len());
-                let mut sum_gas_used = sorter[0].gas_used;
+                let mut sum_gas_used = sorter.first().map(|tx| tx.gas_used).unwrap_or_default();
```

Comment:
this is weird - `gas_used` is just set to `header.gas_used` but we have access to the header here, so we should be able to just do `header.gas_used`?

Reply:
right,

but tbh I don't understand any of this.

Reply:
when a block is empty geth uses an "empty row" https://github.com/ethereum/go-ethereum/blob/8c288b528d3c9653c84321387e78687ed02134d7/eth/gasprice/feehistory.go#L100-L107

So I'm not sure `header.gas_used` would be consistent with the geth impl

Reply:
as an aside geth uses the receipt for the gas used when a block is _not_ empty:
https://github.com/ethereum/go-ethereum/blob/8c288b528d3c9653c84321387e78687ed02134d7/eth/gasprice/feehistory.go#L112

So I wonder if we're even consistent there

Reply:
perhaps this should even be `tx.gas_used` but idk

Reply:
> 

this is only for the reward percentile thing, if the block is empty you are supposed to return 0s for all percentiles. you still return the base fee and gas ratio as normal tho